calico/etcddriver/driver.py

```diff
@@ -305,6 +308,16 @@ def _resync_and_merge(self):
                 self._resync_requested = False
         _log.info("Stop event set, exiting resync loop.")
 
+    def _rotate_etcd_url(self):
```

Comment:
I think this gets called both from the main thread and the watcher thread.  Since those are real threads, I think we need locking around the `_etcd_...` fields.  

Previously they were shared but not mutable.

Alternative would be to have the state owned by the main thread and to pass the base URL to the watcher when starting it so that it's not shared any more.